src/agentscope/agents/agent.py

```diff
@@ -15,16 +16,102 @@
 from agentscope.memory import TemporaryMemory
 
 
-class _RecordInitSettingMeta(ABCMeta):
-    """A wrapper to record the init args into `_init_settings` field."""
+class _AgentMeta(ABCMeta):
+    """The meta-class for agent.
+
+    1. record the init args into `_init_settings` field.
+    2. register class name into `registry` field.
+    """
+
+    def __init__(cls, name: Any, bases: Any, attrs: Any) -> None:
+        if not hasattr(cls, "registry"):
+            cls.registry = {}
+        else:
+            if name in cls.registry:
+                logger.warning(
+                    f"Agent class with name [{name}] already exists.",
+                )
+            else:
+                cls.registry[name] = cls
+        super().__init__(name, bases, attrs)
 
     def __call__(cls, *args: tuple, **kwargs: dict) -> Any:
+        if len(args) > 0:
+            logger.warning(
```

Comment:
When is this function called, and does this warning means we must initialize all agents (including centralized and distributed agents) with keyword arguments?

Reply:
Only agents that need to use `to_dist=True` have this requirement.